src/main/java/com/jakub/todoSandbox/controller/TodoController.java

```diff
@@ -0,0 +1,71 @@
+package com.jakub.todoSandbox.controller;
+
+import com.jakub.todoSandbox.jooq.sample.model.tables.records.TodoRecord;
+import com.jakub.todoSandbox.model.Step;
+import com.jakub.todoSandbox.model.Todo;
+import com.jakub.todoSandbox.model.ValidationException;
+import com.jakub.todoSandbox.repository.TodoRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+import java.util.Optional;
+
+@RestController
+@RequestMapping("/todos")
+public class TodoController {
+
+    @Autowired
+    private final TodoRepository todoRepository;
+
+    public TodoController(TodoRepository todoRepository) {
+        this.todoRepository = todoRepository;
+    }
+
+    @GetMapping
+    public ResponseEntity<List<Todo>> getAllTodos() {
+        return new ResponseEntity<List<Todo>>(todoRepository.findAllTodos(), HttpStatus.OK);
+    }
+
+    @GetMapping("/{id}")
+    public Optional<Todo> getTodo(@PathVariable("id") Long id) {
+        return todoRepository.findTodoById(id);
+    }
+
+    @PostMapping
+    @ResponseStatus(HttpStatus.CREATED)
+    public Todo createTodo(@RequestBody Todo todo) throws ValidationException {
```

Comment:
Make `ValidationException extends RuntimeException` and remove it from the signature of these methods